lib/build/ProjectBuilder.js

```diff
@@ -165,10 +172,18 @@ class ProjectBuilder {
 			return filterProject(projectName);
 		});
 
-		if (this._buildContext.getBuildConfig().createBuildManifest && requestedProjects.length > 1) {
-			throw new Error(
-				`It is currently not supported to request the creation of a build manifest ` +
-				`while including any dependencies into the build result`);
+		if (requestedProjects.length > 1) {
+			const {createBuildManifest, outputStyle} = this._buildContext.getBuildConfig();
+			if (createBuildManifest) {
+				throw new Error(
+					`It is currently not supported to request the creation of a build manifest ` +
+					`while including any dependencies into the build result`);
+			}
+			if (outputStyle === OutputStyleEnum.Flat) {
+				throw new Error(
+					`It is currently not supported to request flat build output ` +
+					`while including any dependencies into the build result`);
```

Comment:
Note: this is explicit enforcement of "Flat" building and is not the same as "Default".

Try, for example, to build the `openui5-sample-app`. If we build it with "Default" flag, then the App is indeed flat, but its dependencies go into the namespaced `/resources/` folder.

However, if we remove the check here (or make an exception for apps), then the build result would be entirelly different- it would flatten the app along with its dependencies into the `dist` folder.

What do you think? How should we handle this?

Reply:
cc @RandomByte

Reply:
My understanding was that `Default` and `Flat` are equal for type `application`. And `Default` just means something else, depending on the type.

Reply:
I came up with this table, which should help us define a consistent solution:

| Requested Output Style / Project Type | Resulting Style
|---|---|
| **Default** |
| `application` | Root project is written `Flat`-style, dependencies in their respective `Default`-style
| `library` | Root project is written `Namespace`-style, dependencies in their respective `Default`-style
| `theme-library` | Root project is written in the style of the sources (multiple namespaces), dependencies in their respective `Default`-style
| `module` | Root project is written with the [configured paths](https://sap.github.io/ui5-tooling/stable/pages/Configuration/#available-path-mappings), dependencies in their respective `Default`-style
| **Flat** |
| `application` | Same as `Default`
| `library` | Root project is written `Flat`-style (without it's namespace, logging warnings for resources outside of it), dependencies in their respective `Default`-style
| `theme-library` | **Unsupported** since a theme-library almost always has more than one namespace
| `module` | **Unsupported** since modules have explicit path mappings configured and no namespace concept
| **Namespace** |
| `application` | Root project is written `Namespace`-style (resources are prefixed with the project's namespace), dependencies are written in their respective `Default`-style
| `library` | Same as `Default`
| `theme-library` | **Unsupported** since a theme-library almost always has more than one namespace
| `module` | **Unsupported** since modules have explicit path mappings configured and no namespace concept

I think we should consider allowing to include dependencies when setting `output-style`. Some projects always require dependencies, like for example third-party modules. Those projects would be prevented from using the new option.

The important thing I realized is that the requested output style **must only apply to the root project**. Dependencies should always use their `Default` style. This is because some project types can't handle the other output-styles as described above.

Reply:
Sounds good to me. Especially, as this still allows to run a build and include a dependency (`--include-dependency`) with type `module` that e.g. adds third-party code to the build output.